tests/test_geography.py

```diff
@@ -68,32 +126,79 @@ def test_polygon_normalize() -> None:
     assert repr(poly_cw) == "POLYGON ((2 0, 2 2, 0 2, 0 0, 2 0))"
 
 
+def test_collection() -> None:
+    objs = [
+        spherely.point(0, 0),
+        spherely.linestring([(0, 0), (1, 1)]),
+        spherely.polygon([(0, 0), (1, 0), (1, 1)]),
+    ]
+
+    coll = spherely.geography_collection(objs)
+
+    assert coll.dimensions == -1
+    assert coll.nshape == 3
+    assert repr(coll).startswith("GEOMETRYCOLLECTION (")
+
+    # TODO: more robust test than using the WKT repr
+    assert repr(coll).count("POINT") == 1
+    assert repr(coll).count("LINESTRING") == 1
+    assert repr(coll).count("POLYGON") == 1
+
+    # TODO: test objects are copied
+    # (for now only test that original objects are preserved)
+    assert [o.nshape for o in objs] == [1, 1, 1]
+
+    # test nested collection
+    # coll2 = spherely.geography_collection(objs + [coll])
```

Comment:
This is not yet supported? (does that require custom handling? FWIW I would be fine with not supporting this)

Reply:
It used to work but not anymore after refactoring Geography (I get a segfault locally).

Reply:
Fixed.